src/fu-engine.c

```diff
@@ -5017,6 +5020,188 @@ fu_engine_get_host_machine_id (FuEngine *self)
 	return self->host_machine_id;
 }
 
+static void
+fu_engine_add_hsi_attrs_tainted (FuEngine *self, GPtrArray *attrs)
+{
+	FwupdHsiAttr *attr;
+
+	/* everything is fine */
+	if (!self->tainted)
+		return;
+
+	/* add runtime attribute */
+	attr = fwupd_hsi_attr_new ("org.fwupd.Hsi.PluginsTainted");
+	fwupd_hsi_attr_set_name (attr, "fwupd daemon");
+	fwupd_hsi_attr_set_summary (attr, "The daemon has been tainted by a non-upstream plugin");
+	fwupd_hsi_attr_add_flag (attr, FWUPD_HSI_ATTR_FLAG_RUNTIME_ISSUE);
+	g_ptr_array_add (attrs, attr);
+}
+
+static void
+fu_engine_add_hsi_attrs_supported (FuEngine *self, GPtrArray *attrs)
+{
+	FwupdHsiAttr *attr;
+	FwupdRelease *rel_newest = NULL;
+	guint64 now = (guint64) g_get_real_time () / G_USEC_PER_SEC;
+	g_autoptr(FuDevice) device = NULL;
+	g_autoptr(GPtrArray) releases = NULL;
+	g_autoptr(GError) error_local = NULL;
+
+	/* find out if there is firmware less than 6 months old */
```

Comment:
you changed code to 12, this should also be 12

Reply:
Fixed.